### PR TITLE
 [Demangle] Check for old-style mangling in getObjCClassByMangledName

### PR DESCRIPTION
**Pull request by harlanhaskins**
This caused an issue where the runtime was unable to find subclasses of resilient subclasses of NSObject until they were first registered by their sugared name with NSClassFromString or were instantiated directly.

rdar://48892003

### CODE REVIEW

File: validation-test/StdlibUnittest/ChildProcessShutdown/RequireOwnProcess.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+var TestSuiteRequireNewProcess = TestSuite("TestSuiteRequireNewProcess")
+
+TestSuiteRequireNewProcess.test("RequireOwnProcess")
+  .requireOwnProcess()
+  .code {
+  Globals.childProcessWasExecuted = true
+}
+
+TestSuiteRequireNewProcess.test("ShouldNotReusePreviousProcess") {
+  expectEqual(false, Globals.childProcessWasExecuted)
```
Comment (jrose-apple) [inline validation-test/StdlibUnittest/ChildProcessShutdown/RequireOwnProcess.swift]: Nitpick: You should test this in both directions by having a test *before* `RequireOwnProcess` too.

File: test/Interpreter/SDK/objc_getClass.swift [hunk 2]
```diff
@@ -195,6 +220,22 @@ testSuite.test("ResilientField")
   expectEqual(ResilientFieldSubclassObjC().subvalue.i, 4)
 }
 
+testSuite.test("ResilientNSObject")
+  .skip(.custom({ getClassHookMissing },
```
Comment (jrose-apple) [inline test/Interpreter/SDK/objc_getClass.swift]: Any reason why this one isn't `requireOwnProcess`?

**Comment by jrose-apple** [comment]
> This caused an issue where the runtime was unable to find subclasses of resilient subclasses of NSObject until they were first registered by their sugared name with NSClassFromString.

This isn't technically correct; they can be registered by NSClassFromString, or by being used directly from Swift.

**Comment by jrose-apple** [comment]
(The actual change looks good.)

File: test/Interpreter/SDK/objc_getClass.swift (hunk 2 above)
Comment (harlanhaskins) [reply test/Interpreter/SDK/objc_getClass.swift]: Oh, how did I miss that?

File: validation-test/StdlibUnittest/ChildProcessShutdown/RequireOwnProcess.swift [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+//
+
+enum Globals {
+  static var globalVariable = 0
+}
+
+var TestSuiteRequireNewProcess = TestSuite("TestSuiteRequireNewProcess")
+
+TestSuiteRequireNewProcess.test("RequireOwnProcessBefore")
+  .requireOwnProcess()
```
Comment (jrose-apple) [inline validation-test/StdlibUnittest/ChildProcessShutdown/RequireOwnProcess.swift]: I meant *not* having this one marked `requireOwnProcess`, to test that your "next" checking works.

**Review by slavapestov — APPROVED** [review]

**Review by jrose-apple — APPROVED** [review]

**Comment by harlanhaskins** [comment]
@swift-ci please test and merge

**Comment by harlanhaskins** [comment]
@swift-ci please test

**Comment by harlanhaskins** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
This broke the Windows build!

**Comment by harlanhaskins** [comment]
@compnerd Oops! I see the issue on your Azure bot. I’ll address it today.

**Pull request merged**